test/puppetlabs/trapperkeeper/services/status/status_core_test.clj

```diff
@@ -15,4 +15,17 @@
 
       (is (nil? (schema/check ServicesInfo @status-fns)))
       (is (= 2 (count (get @status-fns "foo"))))
-      (is (= 1 (count (get @status-fns "bar")))))))
+      (is (= 1 (count (get @status-fns "bar")))))
+
+    (testing (str "registering a service status callback function with a "
+                  "version that already exists causes an error")
+      (is (thrown? IllegalStateException
+                   (update-status-context status-fns "foo"
+                                          "1.1.0" 2 (fn [] "foo repeat")))))
+
+    (testing (str "registering a service status callback function with a "
+                  "status version that already exists but a different service "
+                  "version causes an error")
+      (is (thrown? IllegalStateException
+                   (update-status-context status-fns "foo"
+                                          "1.2.0" 2 (fn [] "foo repeat")))))))
```

Comment:
Ah, I see. I was completely misreading what you were suggesting (I was thinking you were saying not more than one status function registered per service-version, but really what you were saying is no matter how many status functions are registered by a service, they must all have the same service-version). That does seem like it would be really bizarre, and I can see throwing an error for that.